### PR TITLE
Fixes #94 add additional plots

### LINKED ISSUE #94: Report: Additional section for plots
New section should be added to the validation report: _Plots of **successful** simulations_
For every **successful** simulation: add simulation name and plots of few predefined curves in lin and log scale.

tbd. List of predefined curves.

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

File: src/InstallationValidator.Core/Constants.cs [hunk 1]
```diff
@@ -13,6 +16,19 @@ public static class Constants
       public static readonly string DIMENSION_FILE = "OSPSuite.Dimensions.xml";
       public static readonly string DEFAULT_SKIN = "Metropolis"; //"Office 2013 Light Gray";
       public const int BUTTON_HEIGHT = 48;
+      public static readonly string CONCENTRATION = "Concentration in container";
+
+      public static IReadOnlyList<string> PREDEFINED_OUTPUT_PATHS = new[]
+      {
+         concentrationFor("PeripheralVenousBlood", "Plasma (Peripheral Venous Blood)"),
```
Comment (msevestre) [inline src/InstallationValidator.Core/Constants.cs]: List of predefined curves that will be dumped even for valid simulations

File: src/InstallationValidator.Core/Domain/ComparisonSettings.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+﻿using System.Collections.Generic;
+using InstallationValidator.Core.Assets;
+
+namespace InstallationValidator.Core.Domain
```
Comment (msevestre) [inline src/InstallationValidator.Core/Domain/ComparisonSettings.cs]: That object contains all settings for the comparison

File: src/InstallationValidator.Core/Domain/PointwiseComparisonStrategy.cs [hunk 3]
```diff
@@ -3,28 +3,51 @@
...
+
+      private static OutputComparisonResult validOutputComparison(BatchOutputComparison outputValues1, BatchOutputComparison outputValues2, double deviation, ComparisonSettings comparisonSettings)
+      {
+         if (!shouldGenerateOutputsFor(comparisonSettings, outputValues1.Path))
+            return new OutputComparisonResult(outputValues1.Path, comparisonSettings, ValidationState.Valid)
+            {
+               Deviation = deviation
+            };
+
+         return outputComparisonResultFrom(outputValues1, outputValues2, deviation, comparisonSettings, ValidationState.Valid);
```
Comment (msevestre) [inline src/InstallationValidator.Core/Domain/PointwiseComparisonStrategy.cs]: A valid output is generated when the path matches expectations

File: src/InstallationValidator.Core/Extensions/LoggerPresenterExtensions.cs [hunk 4]
```diff
@@ -5,27 +5,27 @@
 
 namespace InstallationValidator.Core.Extensions
 {
-   public static class ComparisonPresenterExtensions
+   public static class LoggerPresenterExtensions
```
Comment (msevestre) [inline src/InstallationValidator.Core/Extensions/LoggerPresenterExtensions.cs]: @rwmcintosh I have performed the changes requested from previous PR

File: src/InstallationValidator.Core/Reporting/BatchComparisonResultTeXBuilder.cs [hunk 5]
```diff
@@ -21,24 +23,40 @@ public override void Build(BatchComparisonResult comparisonResult, OSPSuiteTrack
...
             new SubParagraph(Assets.Reporting.ComparisonFolder(comparisonResult.FolderPathCaption2)),
             comparisonResult.FolderPath2
          };
 
-         var fileComparisonResults = comparisonResult.FileComparisonResults
-            .OrderByDescending(x => x.State)
-            .ThenBy(x => x.FileName);
-
-         objectsToReport.AddRange(fileComparisonResults);
+         objectsToReport.AddRange(reportSimulations(Assets.Reporting.InvalidSimulations, comparisonResult, ValidationState.Invalid));
```
Comment (msevestre) [inline src/InstallationValidator.Core/Reporting/BatchComparisonResultTeXBuilder.cs]: One sub section for each State

File: src/InstallationValidator.Core/Reporting/InstallationValidationResultReporter.cs [hunk 6]
```diff
@@ -22,20 +22,4 @@ public override IReadOnlyCollection<object> Report(InstallationValidationResult
          };
       }
    }
-
-   public class ComparisonResultReporter : OSPSuiteTeXReporter<BatchComparisonResult>
```
Comment (msevestre) [inline src/InstallationValidator.Core/Reporting/InstallationValidationResultReporter.cs]: Moved it in its own file

File: src/InstallationValidator.Core/Services/OutputComparisonResultToCurveChartMapper.cs [hunk 7]
```diff
@@ -51,18 +51,18 @@ private CurveChart createCurveChartWithScaling(OutputComparisonResult outputComp
...
+      private void addCurve(CurveChart curveChart, DataRepository dataRepository, Color color)
+      {
+         curveChart.AddCurvesFor(dataRepository, x => x.PathAsString, _dimensionFactory, (column, curve) =>
          {
-            curve.Color = Color.CornflowerBlue;
+            curve.Color = color;
             curve.VisibleInLegend = true;
             curve.Name = column.Name;
          });
-         curveChart.AddCurvesFor(dataRepository2, x => x.PathAsString, _dimensionFactory, (column, curve) =>
```
Comment (msevestre) [inline src/InstallationValidator.Core/Services/OutputComparisonResultToCurveChartMapper.cs]: defined a method for that here

File: src/InstallationValidator.Core/Services/OutputComparisonResultToCurveChartMapper.cs [hunk 8]
```diff
@@ -30,11 +30,11 @@ public IEnumerable<CurveChart> MapFrom(OutputComparisonResult outputComparisonRe
          var dataRepository1 = _outputResultToDataRepositoryMapper.MapFrom(outputComparisonResult.Output1);
          var dataRepository2 = _outputResultToDataRepositoryMapper.MapFrom(outputComparisonResult.Output2);
 
-         var linearCurveChart = createCurveChartWithScaling(outputComparisonResult, Scalings.Linear, dataRepository1, dataRepository2);
-         var logCurveChart = createCurveChartWithScaling(outputComparisonResult, Scalings.Log, dataRepository1, dataRepository2);
+         yield return createCurveChartWithScaling(outputComparisonResult, Scalings.Log, dataRepository1, dataRepository2);
 
-         addToCurveChart(logCurveChart, dataRepository1, dataRepository2);
```
Comment (msevestre) [inline src/InstallationValidator.Core/Services/OutputComparisonResultToCurveChartMapper.cs]: @rwmcintosh That was called another time for log? any reason?

File: src/InstallationValidator.Core/Reporting/OutputComparisonResultTeXBuilder.cs [hunk 9]
```diff
@@ -27,7 +26,6 @@ public override void Build(OutputComparisonResult outputToReport, OSPSuiteTracke
          {
             new SubParagraph($"{Assets.Reporting.OutputPath}: {outputToReport.Path}"),
             ValidationMessageFor(outputToReport),
-            new LineBreak(),
```
Comment (msevestre) [inline src/InstallationValidator.Core/Reporting/OutputComparisonResultTeXBuilder.cs]: LineBreak is added by the ValidationMessageFor. Thaty way, we do not create an empty line break when validation message was empty (creating an error on latex

File: src/InstallationValidator.Core/Services/OutputComparisonResultToCurveChartMapper.cs [hunk 10]
```diff
@@ -30,11 +30,11 @@ public IEnumerable<CurveChart> MapFrom(OutputComparisonResult outputComparisonRe
...
-         return new []{ linearCurveChart, logCurveChart};
+         //only show linear plit if output is invalid
```
Comment (Yuri05) [inline src/InstallationValidator.Core/Services/OutputComparisonResultToCurveChartMapper.cs]: _plit_=_plot_? :-)

File: src/InstallationValidator.Core/Services/OutputComparisonResultToCurveChartMapper.cs (hunk 8 above)
Comment (rwmcintosh) [inline src/InstallationValidator.Core/Services/OutputComparisonResultToCurveChartMapper.cs]: Vestigial

**Pull request merged**